### ISSUE TITLE
[Bug]: Item tasks not working in some cases

### ISSUE THREAD

**Issue by aaronhowser1** [FTBTeam/FTB-Mods-Issues#534]
### Mod

FTB Quests + Item Filters

### Mod version

ftb-quests-forge-1802.3.9-build.130

### Forge / Fabric version

Forge 1.18.2 40.1.80

### What issue are you having?

This OR filter doesn't work:
![image](https://user-images.githubusercontent.com/15698722/200939076-1852b337-0ed4-4062-9fb9-e13dfa1fa908.png)
![image](https://user-images.githubusercontent.com/15698722/200938592-ba135fc8-2586-4f2d-a814-9356280e7cae.png)

This Pickaxe doesn't work:
![image](https://user-images.githubusercontent.com/15698722/200939207-a8bfd7a8-d8cd-4a9e-9646-d863e1854250.png)
This was made using the pickaxe that I'm holding, it's identical. It has the `#itemfilters:match_nbt` tag.

An `item AND nbt(weak)` also does not work
![image](https://user-images.githubusercontent.com/15698722/200939274-f6f2d553-e6cf-4048-9ced-c8b60ad8140f.png)

### Steps to reproduce

n/a

**Comment by desht** [comment]
Can't reproduce any of this.

I don't have those mods in my dev workspace, but I have tested with vanilla items:

1. OR filter with diamond,emerald,lapis,redstone triggers the item task correctly if I put any of those 4 items in my inventory. The gems in your example are just simple items (no NBT), right?
2. Enchanted pickaxe with 3 enchantments (and "Match NBT" set to true in the item task properties) triggers only when that exact pickaxe is picked up, as expected.

> It has the #itemfilters:match_nbt tag.

That's the wrong tag; the tag is `itemfilters:check_nbt`

**Comment by desht** [comment]
For the third example:
* What _exactly_ is in your "NBT Filter (weak)" ?  copy/paste of the string please
* Is NBT matching true/false/default for the item task in question? (should be false if you're using an AND + NBT filter)

**Comment by desht** [comment]
Finally make sure you are on the latest Item Filters release (1802.2.8-build.47)

That should be required by this version of FTBQ but isn't - I'll fix that.

**Comment by aaronhowser1** [comment]
> OR filter with diamond,emerald,lapis,redstone triggers the item task correctly if I put any of those 4 items in my inventory. The gems in your example are just simple items (no NBT), right?

No, they have NBT. They aren't in the creative menu, they're randomly generated through a command

**Comment by desht** [comment]
I just tested Inferno 1.2.1 and dropped in the most recent FTB Quests (1802.3.10-build.138), and it correctly matched that flint pickaxe.  Whereas 1802.3.9-build.130 did not match the pick.

I'm puzzled as to why the newer version works, since there aren't item task related changes between the two version (latest version adds the reward blocking feature, loot crate opener and a couple of other unrelated minor changes).  But it works...
